### PR TITLE
Adds pangolin 4.1.2 and pdata 1.12

### PR DESCRIPTION
**Pull request by kapsakcj**
This PR adds a dockerfile for pangolin 4.1.2  and pangolin-data v1.12 https://github.com/cov-lineages/pangolin-data/releases/tag/v1.12

This corresponds with pango-designation v1.12: https://github.com/cov-lineages/pango-designation/releases/tag/v1.12

~~It currently does not build successfully, since the corresponding pangolin-assignment cache for v1.12 has not yet been released. It should build fine once that release is made: https://github.com/cov-lineages/pangolin-assignment/releases~~

Setting as a draft PR for now, will change when the docker image builds successfully and is ready for review.

New lineages capable of being detected with this new version (list pulled from pango-designation 1.12 release notes):

```
BA.2.38.1
BA.2.74
BA.2.75
BA.2.76
BA.2.77
BA.2.78
BA.2.79
BA.2.79.1
BA.2.80

BA.4.1.3
BA.4.1.4
BA.4.2
BA.4.3
BA.4.4
BA.4.5
BA.4.6

BA.5.1.1
BA.5.1.2
BA.5.1.3
BA.5.1.4

BA.5.2.2
BA.5.2.3
BA.5.2.4

BA.5.3.3
BA.5.3.4

BA.5.6

BE.1.1
BE.2
BE.3

BF.1.1
BF.2
BF.3
BF.4
BF.5

BG.3
BG.4
```

Pull Request (PR) checklist:
- [x] Include a description of what is in this pull request in this message.
- [x] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [x] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [x] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [x] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [x] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [x] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [x] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [x] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

### CODE REVIEW

**Comment by kapsakcj** [comment]
Docker image builds successfully now!

I also added in a test on a BA.1 virus from Florida (thank you @SESchmedes & team for submitting your data to NCBI 👍 )

**Comment by kapsakcj** [comment]
output of `pangolin --all-versions` as reported in GHA workflow log:

```
pangolin: 4.1.2
pangolin-data: 1.12
constellations: v0.1.10
scorpio: 0.3.17
pangolin-assignment: 1.12
usher 0.5.6
gofasta 1.1.0
minimap2 2.24-r1122
faToVcf: 426
```

File: pangolin/4.1.2/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+# test functionality of assignment-cache option
+RUN pangolin --use-assignment-cache /pangolin/pangolin/test/test_seqs.fasta
+
+# download B.1.1.7 genome from Utah
+ADD https://raw.githubusercontent.com/StaPH-B/docker-builds/master/tests/SARS-CoV-2/SRR13957123.consensus.fa /test-data/SRR13957123.consensus.fa
+
+# test on a B.1.1.7 genome
+RUN pangolin /test-data/SRR13957123.consensus.fa --analysis-mode usher -o /test-data/SRR13957123-pusher && \
+ cat /test-data/SRR13957123-pusher/lineage_report.csv
```
Comment (erinyoung) [inline pangolin/4.1.2/Dockerfile:111]: Since these don't require datasets installed, can you have these run before datasets and unzip are added?

**Review by erinyoung — CHANGES_REQUESTED** [review]
It's just a small thing, but could you change the order of the tests. I'd hate to have the tests work only because of something you downloaded in the test phase.

I don't know why it took me so long to catch this.

File: pangolin/4.1.2/Dockerfile (hunk 1 above)
Comment (kapsakcj) [reply pangolin/4.1.2/Dockerfile:111]: sure thing, I will change the order

**Comment by erinyoung** [comment]
I like the snarkiness of the samples in the output. 'This_seq_is_literally_just_N' might be my favorite.

```
#14 16.88 taxon,lineage,conflict,ambiguity_score,scorpio_call,scorpio_support,scorpio_conflict,scorpio_notes,version,pangolin_version,scorpio_version,constellation_version,is_designated,qc_status,qc_notes,note
#14 16.88 India seq,B.1.617.1,0.0,,B.1.617.1-like,1.0,0.0,scorpio call: Alt alleles 11; Ref alleles 0; Amb alleles 0; Oth alleles 0,PUSHER-v1.12,4.1.2,0.3.17,v0.1.10,False,pass,Ambiguous_content:0.02,Usher placements: B.1.617.1(1/1)
#14 16.88 b117,B.1.1.7,0.0,,Alpha (B.1.1.7-like),0.91,0.04,scorpio call: Alt alleles 21; Ref alleles 1; Amb alleles 1; Oth alleles 0,PUSHER-v1.12,4.1.2,0.3.17,v0.1.10,False,pass,Ambiguous_content:0.02,Usher placements: B.1.1.7(4/4)
#14 16.88 outgroup_A,A,0.0,,,,,,PUSHER-v1.12,4.1.2,0.3.17,v0.1.10,False,pass,Ambiguous_content:0.02,Usher placements: A(1/1)
#14 16.88 issue_57_torsten_seq,Unassigned,,,,,,,PUSHER-v1.12,4.1.2,0.3.17,v0.1.10,False,fail,failed to map,
#14 16.88 This_seq_has_6000_Ns_in_18000_bases,Unassigned,,,,,,,PUSHER-v1.12,4.1.2,0.3.17,v0.1.10,False,fail,failed to map,
#14 16.88 This_seq_has_no_seq,Unassigned,,,,,,,PUSHER-v1.12,4.1.2,0.3.17,v0.1.10,False,fail,failed to map,
#14 16.88 This_seq_is_too_short,Unassigned,,,,,,,PUSHER-v1.12,4.1.2,0.3.17,v0.1.10,False,fail,Ambiguous_content:0.9,
#14 16.88 This_seq_has_lots_of_Ns,Unassigned,,,,,,,PUSHER-v1.12,4.1.2,0.3.17,v0.1.10,False,fail,Ambiguous_content:0.98,
#14 16.88 This_seq_is_literally_just_N,Unassigned,,,,,,,PUSHER-v1.12,4.1.2,0.3.17,v0.1.10,False,fail,failed to map,
#14 16.88 Japan_seq,B,0.5,,,,,,PANGO-v1.12,4.1.2,0.3.17,v0.1.10,True,pass,Ambiguous_content:0.02,Assigned from designation hash.
#14 16.88 USA_seq,B.1.314,0.0,,,,,,PANGO-v1.12,4.1.2,0.3.17,v0.1.10,True,pass,Ambiguous_content:0.02,Assigned from designation hash.
#14 16.88 Unassigned_omicron_seq,BA.1,0.0,,Probable Omicron (BA.1-like),0.71,0.08,scorpio call: Alt alleles 42; Ref alleles 5; Amb alleles 9; Oth alleles 3,PUSHER-v1.12,4.1.2,0.3.17,v0.1.10,False,pass,Ambiguous_content:0.03,Usher placements: BA.1(1/1)
#14 DONE 18.4s
```

**Pull request merged**